pkg/config/config.go

```diff
@@ -388,3 +422,106 @@ func checkAdvertiseAddressConfigured(advertiseAddress string) error {
 	}
 	return fmt.Errorf("Advertise address: %s not present in any interface", advertiseAddress)
 }
+
+func validateRouterListenAddress(entries []string, advertiseAddress string, skipInterface bool) error {
+	addresses, err := AllowedListeningIPAddresses()
+	if err != nil {
+		return err
+	}
+	names, err := AllowedNICNames()
+	if err != nil {
+		return err
+	}
+	for _, entry := range entries {
+		if net.ParseIP(entry) != nil {
+			if entry == advertiseAddress && !skipInterface {
+				continue
+			}
+			if !slices.Contains(addresses, entry) {
+				return fmt.Errorf("IP %v not present in the host", entry)
+			}
+		} else if slices.Contains(names, entry) {
+			continue
+		} else {
+			return fmt.Errorf("entry %v not present in the host", entry)
+		}
+	}
+	return nil
+}
+
+func getBannedIPs() ([]*net.IPNet, error) {
+	banned := make([]*net.IPNet, 0)
+	for _, entry := range defaultRouterForbiddenCIDRs {
+		_, netIP, err := net.ParseCIDR(entry)
+		if err != nil {
+			return nil, err
+		}
+		banned = append(banned, netIP)
+	}
+	return banned, nil
+}
+func getHostAddresses() ([]net.IP, error) {
+	handle, err := netlink.NewHandle()
+	if err != nil {
+		return nil, err
+	}
+	links, err := handle.LinkList()
+	if err != nil {
+		return nil, err
+	}
+	addresses := make([]net.IP, 0)
```

Comment:
Is there a reasonable size that we could preallocate?
```suggestion
	addresses := make([]net.IP, 0, len(links) * 2)
```

Reply:
I see the point, but I dont think this will matter much as the code is just ran once at the start, and the amount of total IPs in the system wont get to the point where reallocations hit performance.
Plus, slices are contiguous memory so you might be robbing cache lines from other variables. But again this is negligible with these sizes.
Will do for correctness though.